src/controllers/dishController.js

```diff
@@ -55,8 +54,46 @@ exports.getDishById = async (req, res) => {
       throw new Error('Dish not found');
     }
 
-    res.json(dish);
+    res.json({ dish });
   } catch (error) {
     res.status(500).json({ error: error.message });
   }
 };
+
+exports.dishesRatings = async (req, res) => {
+  const token = req.cookies.jwt;
+  const decodedToken = decodeJwtToken(token);
+
+  const { id } = req.params;
+  const { rating } = req.body;
+
+  try {
+    const dish = await Dish.findById(id);
+
+    const existingRatingIndex = dish.ratings.findIndex((r) =>
+      r.userId.equals(decodedToken.userId)
+    );
+
+    if (existingRatingIndex !== -1) {
+      dish.ratings[existingRatingIndex].rating = rating;
+    } else {
+      const newRating = { userId: decodedToken.userId, rating };
+      dish.ratings.push(newRating);
+    }
+
+    const ratingsCount = dish.ratings.length;
+    const ratingSum = dish.ratings.reduce(
+      (sum, value) => sum + value.rating,
+      0
+    );
+    const averageRating = ratingsCount > 0 ? ratingSum / ratingsCount : 0;
+
+    dish.ratingAve = averageRating.toFixed(1);
+
+    await dish.save();
+
+    res.status(201).json('Dish Rated Successfully');
+  } catch (error) {
+    res.status(500).json(error.message);
+  }
+};
```

Comment:
I think we should move this inside the client controller. We can create new endpoint in client routes such as 
router.post('/dishes/:id/rateDish', clientController.rateDish);
there we can rate a dish and control that;
- only clients can rate the dishes
- a client can rate a dish one time only, cause clients shouldn't be able change the rate of the same dish